code/modules/mob/living/simple_animal/friendly/cat.dm

```diff
@@ -45,6 +45,11 @@
 					movement_target = null
 					stop_automated_movement = 0
 					break
+			for(var/obj/item/toy/cattoy/T in view(1,src))

```

Comment:
First off make this a pick or something of that sort instead of a for unless you want another "The bible melts!"